apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx

```diff
@@ -41,50 +43,57 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
       redirectType: RedirectType.Team,
       eventTypeSlug: meetingSlug,
       currentQuery: context.query,
+      prismaClient: prisma,
     });
 
     if (redirect) {
       return redirect;
     }
   }
 
-  const team = await getTeamWithEventsData(teamSlug, meetingSlug, isValidOrgDomain, currentOrgDomain);
+  const [orgId, booking] = await Promise.all([
+    isOrgContext ? getOrgId(currentOrgDomain) : Promise.resolve(null),
+    rescheduleUid ? getBookingForReschedule(`${rescheduleUid}`, session?.user?.id) : Promise.resolve(null),
+  ]);
 
-  if (!team || !team.eventTypes?.[0]) {
+  const team = await getTeamData(teamSlug, orgId);
+
+  if (!team) {
     return { notFound: true } as const;
   }
 
-  const eventData = team.eventTypes[0];
+  const eventData = await getEventTypeData(meetingSlug, team.id);
+
+  if (!eventData) {
+    return { notFound: true } as const;
+  }
 
   if (rescheduleUid && eventData.disableRescheduling) {
     return { redirect: { destination: `/booking/${rescheduleUid}`, permanent: false } };
   }
 
+  if (
+    booking?.status === BookingStatus.CANCELLED &&
+    !allowRescheduleForCancelledBooking &&
+    !eventData.allowReschedulingCancelledBookings
+  ) {
+    return {
+      redirect: {
+        permanent: false,
+        destination: `/team/${teamSlug}/${meetingSlug}`,
+      },
+    };
+  }
+
   const eventTypeId = eventData.id;
-  const eventHostsUserData = await getUsersData(
-    team.isPrivate,
-    eventTypeId,
-    eventData.hosts.map((h) => h.user)
-  );
   const orgSlug = isValidOrgDomain ? currentOrgDomain : null;
   const name = team.parent?.name ?? team.name ?? null;
 
-  let booking: GetBookingType | null = null;
-  if (rescheduleUid) {
-    booking = await getBookingForReschedule(`${rescheduleUid}`, session?.user?.id);
-    if (
-      booking?.status === BookingStatus.CANCELLED &&
-      !allowRescheduleForCancelledBooking &&
-      !eventData.allowReschedulingCancelledBookings
-    ) {
-      return {
-        redirect: {
-          permanent: false,
-          destination: `/team/${teamSlug}/${meetingSlug}`,
-        },
-      };
-    }
-  }
+  const eventHostsUserData = getEventHosts(
```

Comment:
- renamed `getUsersData` to `getEventHosts` 
- getEventHosts is now synchronous as I removed the unnecessary event-type DB fetch inside it